akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpChallenge.scala

```diff
@@ -22,6 +23,15 @@ final case class HttpChallenge(scheme: String, realm: String,
   def getParams: util.Map[String, String] = params.asJava
 }
 
+// FIXME: Replace these two constructors with a single constructor and default 'params' Map.empty in Akka 3.0
+object HttpChallenge {
+  def apply(scheme: String, realm: Option[String]): HttpChallenge =
+    HttpChallenge(scheme, realm.orNull, Map.empty[String, String])
+
+  def apply(scheme: String, realm: Option[String], params: Map[String, String]): HttpChallenge =
```

Comment:
:+1:

-- 
Konrad `ktoso` Malawski
Akka <http://akka.io> @ Lightbend <http://lightbend.com>

On 27 July 2016 at 21:53:27, Ian Clegg (notifications@github.com) wrote:

In
akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpChallenge.scala
<https://github.com/akka/akka/pull/21043#discussion_r72510619>:

> @@ -22,6 +23,15 @@ final case class HttpChallenge(scheme: String, realm: String,
>    def getParams: util.Map[String, String] = params.asJava
>  }
>
> +// FIXME: Replace these two constructors with a single constructor and default 'params' Map.empty in Akka 3.0
> +object HttpChallenge {
> +  def apply(scheme: String, realm: Option[String]): HttpChallenge =
> +    HttpChallenge(scheme, realm.orNull, Map.empty[String, String])
> +
> +  def apply(scheme: String, realm: Option[String], params: Map[String, String]): HttpChallenge =

Removed the // FIXME

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/akka/akka/pull/21043/files/90e1938884375e7c1562713968dd1bdbc175ce84#r72510619>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/AAHYk4dUMNyBTqVzOT_1z0Pp5pL0oiycks5qZ7c3gaJpZM4JVJez>
.